vic/drivers/image/src/vic_write.c

```diff
@@ -35,92 +35,131 @@
 void
 vic_write(void)
 {
-    extern out_data_struct **out_data;
-    extern domain_struct     global_domain;
-    extern domain_struct     local_domain;
-    extern nc_file_struct    nc_hist_file;
-    extern nc_var_struct     nc_vars[N_OUTVAR_TYPES];
-    extern size_t            current;
-    int                      dimids[MAXDIMS];
-    size_t                   i;
-    size_t                   j;
-    size_t                   k;
-    size_t                   grid_size;
-    size_t                   ndims;
-    double                  *dvar = NULL;
-    size_t                   dcount[MAXDIMS];
-    size_t                   dstart[MAXDIMS];
+    extern out_data_struct   **out_data;
+    extern domain_struct       global_domain;
+    extern domain_struct       local_domain;
+    extern nc_file_struct      nc_hist_file;
+    extern nc_var_struct       nc_vars[N_OUTVAR_TYPES];
+    extern size_t              current;
+    extern global_param_struct global_param;
 
-    grid_size = global_domain.n_ny * global_domain.n_nx;
+    int                        dimids[MAXDIMS];
+    size_t                     i;
+    size_t                     j;
+    size_t                     k;
+    size_t                     grid_size;
+    size_t                     ndims;
+    double                    *dvar = NULL;
+    size_t                     dcount[MAXDIMS];
+    size_t                     dstart[MAXDIMS];
+    int                        nc_time;
+    
+    if ((current +
+         1) %
+        (global_param.model_steps_per_day /
+         global_param.output_steps_per_day) ==
+        0) {
+        // get start location in NetCDF file for current timestep.
+        nc_time =
+            ((current +
+              1) /
+             (global_param.model_steps_per_day /
+         global_param.output_steps_per_day)) - 1;
```

Comment:
I think it would be better, for now, to use the `date2num` and the current `dmy_struct` to come up with the time value.